src/envoy/utils/filter_state_utils.cc

```diff
@@ -25,13 +27,20 @@ using ::Envoy::Router::StringAccessor;
 using ::Envoy::Router::StringAccessorImpl;
 using ::Envoy::StreamInfo::FilterState;
 
+// FilterState container needed to store the google.rpc.Status error proto.
+struct RpcStatusWrapper : public Envoy::StreamInfo::FilterState::Object {
+  google::rpc::Status status_;
```

Comment:
use class, and hike status_ in the private and create a copy constructor.

Reply:
As discussed offline, no need (?)